Update docker-compose and makefile

dirsouza opened the pull request:
### Descrição
Foram feitas as seguintes alterações no projeto:

**Makefile** 

- Adicionado o comando `start-dev--no-cache` para fazer o build do projeto sem cache;
- Editado o comando `start-dev` para fazer o build do projeto utilizando o cache (os containers sobem mais rápido).

**docker-compose.yml**

- Adicionado a tag `labels` no serviço do `mysql`, com as seguintes propriedades:
  - `traefik.enable=true`
    > Permite ao `traefik` monitorar o serviço.
  - `traefik.port=3306`
    > Informa ao `traefik` a porta em quê o serviço está.
  - `traefik.backend=mysql`
    > Informa ao `traefik` como o serviço irá ser acessado internamente.

### Cenário de teste

1. **Testar o build do projeto**:
   1. Execute o comando `make down` para parar os containers;
   2. **_CASO I_** - Já tenha feito o build do projeto anteriormente com o `make start-dev`:
      1. Não é necessário fazer o build do projeto novamente, contudo é necessário o `make down` e depois o `make start-dev`.
   3. **_CASO II_** - Não tenha feito o build do projeto anteriormente:
       1. Necessário fazer o build do projeto com o `make start-dev--no-cache`, posteriormente, não havendo mudanças nas imagens dos containers, somente será necessário o `make start-dev`.
   1. Ao finalizar, os serviços deverão estarem disponíveis para o acesso web.

2. **Acesso ao banco de dados**:
   1. Abra o `phpMyAdmin` pela URL descrita no dashboard do `traefik`, siga os passos descritos abaixo:
      - **Servidor**: _mysql_ ou _<ip interno do serviço>:3306_
      - **Utilizador**: _root_ ou conforme informado no `.env`
      - **Palavra-passe**: _root_ ou conforme informado no `.env`

Resultado esperado: Todos os serviços estarem sendo monitorados pelos `traefik` e com seus devidos hosts de acesso ativos.

Commit:
Removido a exposição literal de portas no serviço do mysql

Commit:
Adicionado target start-dev--no-cache para build sem cache

Commit:
Correção start-dev--no-cache

Commit:
Update das variáveis de 'manaushumana' para 'mmh' e acrescentado a variável TIMEZONE

Commit:
Update README.md

Commit:
Acrescentado Dockerfile para o serviço do MySql

bruno-s-campos commented on Makefile:
```diff
@@ -1,5 +1,8 @@
+start-dev: front back dev up-dev
+
+.PHONY: start-dev--no-cache
+start-dev--no-cache: front back dev build-no-cache up-dev
```
Sugestão: para manter o padrão, eu mudaria para **start-dev-no-cache**.

dirsouza replied:
Okay

bruno-s-campos commented on Makefile:
```diff
@@ -1,5 +1,8 @@
 .PHONY: start-dev
-start-dev: front back dev build-no-cache up-dev
+start-dev: front back dev up-dev
```
Teria que ter um build também:
```
start-dev: front back dev build up-dev
```
Se não fazer o build do projeto, não ocorre a instalação do Laravel.

dirsouza replied:
Okay

bruno-s-campos commented on .env-example:
```diff
@@ -14,7 +14,8 @@ TRAEFIK_AUTH=
 MYSQL_USER=root
 MYSQL_PASSWORD=root
 MYSQL_ROOT_PASSWORD=root
+TIMEZONE=America/Manaus
```
Não seria melhor deixar o timezone em UTC?

Podemos deixar qualquer operação de data no lado do backend em UTC, e o front ficar responsável de fazer a conversão para o timezone local do usuário.

dirsouza replied:
Okay

bruno-s-campos commented on README.md:
```diff
@@ -34,18 +32,22 @@ Estrutura em containers usando o [Traefik v1.7](https://docs.traefik.io/v1.7), t
-	# modo: desenvolvimento em modo detach (em segundo plano)
-	./compose.sh "up -d" dev
+    # Demais builds
+    make start-dev
```
O `make start-dev` ou `make start-dev--no-cache` só precisa executar uma vez (a não ser que tenha alteração nas dependências do projeto, que nesse caso é melhor executar `make down start-dev`). Nos demais builds, basta executar `make up`.

dirsouza replied:
Okay

Commit:
Removido `start-dev--no-cache`

Sendo o `start-dev` o primeiro build e `up-dev` para subir os containers, é desnecessário o `start-dev--no-cache`

Commit:
Substituído o America/Manaus por UTC

Commit:
Atualização do README.md
